kore/src/Kore/Log/DebugAppliedRewriteRules.hs

```diff
@@ -32,8 +33,9 @@ import Pretty (
 import Pretty qualified
 
 data DebugAppliedRewriteRules = DebugAppliedRewriteRules
-    { configuration :: Pattern VariableName
-    , appliedRewriteRules :: [SourceLocation]
+    { configuration :: !(Pattern VariableName)
+    , ruleLabels :: ![Text]
+    , appliedRewriteRules :: ![SourceLocation]
```

Comment:
Wouldn't this be a sum type of lables and sourcelocation? Basically, it should be `ruleId :: RuleId` where `RuleId = Label | SourceLocation`. Does that make sense or is there some detail I'm overlooking?

Reply:
I think it doesn't fit. Label and source location are used both. Label is used for selecting which debug message should be printed. And source location is used to print where the rule is placed in the source definition, like this:
```
The rules at following locations were applied:
 /home/alexey/Projects/RV/imp/imp.k:41:20-41:91
```

Reply:
Hmm, I think labels should be preferred over source locations. If a rule has a label, print the label, if it doesn't, print the source location

Reply:
Here is the output for option `--debug-attempt-equation`:
```
kore-exec: [237509] Debug (DebugAttemptEquation):
    applying equation at /home/alexey/Projects/RV/equational-rules/functions.k:21:20-21:59 to term:
        /* Fl Fn D Spa */
        Lbllength'LParUndsRParUnds'FUNCTIONS'Unds'Int'Unds'ConsList{}(
            /* Fl Fn D Sfa */ RuleVarLIST:SortConsList{}
        )
Context:
    (DebugAttemptEquation) while applying equation at /home/alexey/Projects/RV/equational-rules/functions.k:21:20-21:59
kore-exec: [239006] Debug (DebugAttemptEquation):
    equation is not applicable:
    equation did not match term:  unimplemented matching case
Context:
    (DebugAttemptEquation) while applying equation at /home/alexey/Projects/RV/equational-rules/functions.k:21:20-21:59
kore-exec: [245011] Debug (DebugAttemptEquation):
    applying equation at /home/alexey/Projects/RV/equational-rules/functions.k:21:20-21:59 to term:
        /* Fl Fn D Spa */
        Lbllength'LParUndsRParUnds'FUNCTIONS'Unds'Int'Unds'ConsList{}(
            /* Fl Fn D Sfa */ ConfigVarLIST:SortConsList{}
        )
Context:
    (InfoReachability) while simplifying the claim
        (DebugAttemptEquation) while applying equation at /home/alexey/Projects/RV/equational-rules/functions.k:21:20-21:59
kore-exec: [246091] Debug (DebugAttemptEquation):
    equation is not applicable:
    equation did not match term:  unimplemented matching case
Context:
    (DebugAttemptEquation) while applying equation at /home/alexey/Projects/RV/equational-rules/functions.k:21:20-21:59
        (InfoReachability) while simplifying the claim
#Top
```
Here we see source location for every attempt, and don't see any label information. Label is used only to select which debug messages should be printed. In similar way we get debug output for `--debug-attempt-rewrite` option:
```
kore-exec: [330673] Debug (DebugAttemptedRewriteRules):
    The rule at following location was attempted:
    /home/alexey/Projects/RV/imp/imp.k:113:21-113:73
    On configuration:
      \and{SortGeneratedTopCell{}}(
          /* term: */
          /* Fl Fn D Sfa */
          Lbl'-LT-'generatedTop'-GT-'{}(
              /* Fl Fn D Sfa Cl */
              Lbl'-LT-'T'-GT-'{}(
                  /* Fl Fn D Sfa Cl */
                  Lbl'-LT-'k'-GT-'{}(
                      /* Fl Fn D Sfa Cl */
                      kseq{}(
...
```
My changes doesn't affect the debug output format. I only added options an its selection by label. Therefore I think that if we want to change the debug output and to add to it the selected label, then we need to open a separate issue for this.

Reply:
Ok, thank you for investigating. Will do!